src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -186,34 +153,16 @@ class TaskReplaceActor(
 
     case Done => // This is the result of successful launchQueue.addAsync(...) call. Nothing to do here
 
-    case CheckFinished => checkFinished()
-
-  }
-
-  override def instanceConditionChanged(instanceId: Instance.Id): Unit = {
-    if (healthyInstances(instanceId) && readyInstances(instanceId)) killNextOldInstance(Some(instanceId))
-    checkFinished()
-  }
+    case CheckFinished => // Notging to do, this is done in step()
 
-  def reconcileAlreadyStartedInstances(): Unit = {
-    logger.info(s"Deployment $deploymentId: Reconciling instances during ${runSpec.id} deployment: found ${instancesAlreadyStarted.size} already started instances " +
-      s"and ${oldActiveInstanceIds.size} old instances: ${if (currentInstances.size > 0) currentInstances.map{ i => i.instanceId -> i.state.condition } else "[]"}")
-    instancesAlreadyStarted.foreach(reconcileHealthAndReadinessCheck)
   }
 
-  // Careful not to make this method completely asynchronous - it changes local actor's state `instancesStarted`.
-  // Only launching new instances needs to be asynchronous.
-  def launchInstances(): Future[Done] = {
-    val leftCapacity = math.max(0, ignitionStrategy.maxCapacity - oldActiveInstanceIds.size - instancesStarted)
-    val instancesNotStartedYet = math.max(0, runSpec.instances - instancesStarted)
-    val instancesToStartNow = math.min(instancesNotStartedYet, leftCapacity)
+  def launchInstances(instancesToStartNow: Int): Future[Done] = {
     if (instancesToStartNow > 0) {
-      logger.info(s"Deployment $deploymentId: Restarting app $pathId: queuing $instancesToStartNow new instances")
-      instancesStarted += instancesToStartNow
+      logger.info(s"Deployment $deploymentId: Restarting app $pathId: queuing $instancesToStartNow new instances.")
       launchQueue.add(runSpec, instancesToStartNow)
     } else {
-      logger.info(s"Deployment $deploymentId: Restarting app $pathId. No need to start new instances right now with leftCapacity = $leftCapacity, " +
-        s"instancesNotStartedYet = $instancesNotStartedYet and instancesToStartNow = $instancesToStartNow")
+      logger.info(s"Deployment $deploymentId: Restarting app $pathId. No need to start new instances right now.")
```

Comment:
I'm not sure this message is still valid. It is likely that one of the launchCommand invokation will have instancesToStartNow==0 (if the deployment is stalled for instance, or if there is no task to launch because we are at the max over capacity). In this case we would display a message saying "we are just restarting the app" which is weird.